src/components/state/renderApp.tsx

```diff
@@ -3,12 +3,27 @@ import * as ReactDOM from "react-dom";
 
 import { Site } from "../Site";
 import { rootSiteElement } from "./rootSiteElement";
-import { newSitePropsFromState } from "./newSiteProps";
+import { sitePropsFromState } from "./sitePropsFromState";
+import { updateStateToUseOnBackAndForwardNav } from "../nav/updateStateToUseOnBackAndForwardNav";
+import { FieldBinder } from "../forms/types/FieldBinder";
+import { Dictionary } from "lodash";
+import { mapProperties } from "../../utils/list";
 
 export const renderApp = (state) => {
     window.onresize = () => renderApp(state)
+    updateStateToUseOnBackAndForwardNav(state)
+    
+    const bindField = (fieldBinder:FieldBinder<any,any>) => fieldBinder(state, renderApp)
```

Comment:
this is how we provide children with the ability to bind one or several fields to the current state, ability to re-render. we pass this bindField / bindFields function into the props of children, where children may choose to use them to wire up some unbound fields they care about to the state.